src/scripts/main.js

```diff
@@ -1,3 +1,279 @@
 'use strict';
 
-// write your code here
+let board;
+let score = 0;
+const rows = 4;
+const columns = 4;
+let isFirstMove = true;
+let changed = {
+  left: true,
+  right: true,
+  up: true,
+  down: true,
+};
+
+const startMessage = document.querySelector('.message-start');
+const winMessage = document.querySelector('.message-win');
+const loseMessage = document.querySelector('.message-lose');
+const startButton = document.querySelector('.start');
+
+window.onload = function() {
+  setGame();
+};
+
+document.addEventListener('keyup', (e) => {
+  switch (e.code) {
+    case 'ArrowLeft':
+      slideLeft();
+      break;
+    case 'ArrowRight':
+      slideRight();
+      break;
+    case 'ArrowUp':
+      slideUp();
+      break;
+    case 'ArrowDown':
+      slideDown();
+      break;
+    default:
+      break;
+  }
+
+  const isAnythingChanged = !!Object.values(changed)
+    .filter(value => value).length;
+
+  if (!isAnythingChanged && !isFreeSpace()) {
+    loseMessage.classList.remove('hidden');
+
+    return;
+  }
+  setTimeout(() => addNumber(), 300);
+});
+
+startButton.addEventListener('click', e => {
+  resetBoard();
+  startGame();
+});
+
+const startGame = () => {
+  addNumber();
+  addNumber();
+  startMessage.classList.add('hidden');
+  winMessage.classList.add('hidden');
+  loseMessage.classList.add('hidden');
+  startButton.classList.replace('start', 'restart');
+  startButton.innerHTML = 'Restart';
+};
+
+const resetBoard = () => {
+  board = [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+  ];
+
+  board.forEach((row, rowIndex) => {
+    row.forEach((cell, columnIndex) => {
+      const tile = document.getElementById(`${rowIndex}-${columnIndex}`);
+
+      updateTile(tile, 0);
+    });
+  });
+};
+
+const setGame = () => {
+  board = [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+  ];
```

Comment:
consider  to create constant for initial board, as it uses at least two times

Reply:
Thanks, done)